Inlay hints for types of consts

nickysn opened the pull request:
This adds inlay hint support for the types of consts.

nickysn committed:
Show inlay hints for types of const declarations

nickysn committed:
Don't show inlay hints for const with explicitly declared type

nickysn committed:
In const inlay type hints, convert the "int literal(XXX)" type name to "int"

Araq commented on compiler/suggest.nim:
```diff
@@ -251,11 +251,18 @@ proc `$`*(suggest: Suggest): string =
+      "int"
+    else:
+      s
+
```
That's a hack. Instead tell `typeToString` to ignore the literal or something like that.

nickysn replied:
Good idea. I introduced a new value preferInlayHint and removed the hack.

nickysn committed:
Introduced preferInlayHint as a new prefer type value to typeToString(). Use it
for forming the type names for inlay hints, instead of doing a hack that checks
whether the type name starts with "int literal(".
